maze_transformer/evaluation/plot_maze.py

```diff
@@ -31,44 +15,193 @@ class PathFormat:
+        """
+        if label is None:
+            label = "predicted path {path_num}".format(
+                path_num=len(self.predicted_paths) + 1
+            )
+        if color is None:
+            color = self.DEFAULT_PREDICTED_PATH_COLORS[len(self.predicted_paths)]
+
+        self.predicted_paths.append(
+            PathFormat(
+                path=path,
+                label=label,
+                fmt=fmt,
+                color=color,
+                line_width=line_width,
+                quiver_kwargs=quiver_kwargs,
+            )
+        )
+        return self
+
+    def _latticemaze_to_img(
+        self, unit_length: int = UNIT_LENGTH
+    ) -> NDArray["row col", bool]:
+        """
+        Build an image to visualise the maze.
+        Each "unit" consists of a node and the right and lower adjacent wall/connection. Its area is ul * ul.
+        - Nodes are displayed as white squares of area: (ul-1) * (ul-1)
+        - Walls are displayed as black recktangles of area: 1 * (ul-1)
+        - Connections are displayed as light grey or white rectangles of area: 1 * (ul-1); color is depending on show_connections argument
+
+        Axes definition:
+        (0,0)     y, col
+        ----|----------->
+        x,  |
+        row |
+            |
+            v
+
+        Returns a matrix of side length (ul) * n + 1 where n is the number of nodes.
+        """
+        # Set color of connections (using plt.imshow(cmap='grey))
+        node_color = 1
+        connection_color = node_color * 0.93
+
+        # Set up the background (walls everywhere)
+        img: NDArray["row col", int] = np.zeros(
+            (
+                self.maze.grid_shape[0] * unit_length + 1,
+                self.maze.grid_shape[1] * unit_length + 1,
+            ),
+            dtype=float,
+        )
+
+        # Draw nodes and connections by iterating through lattice
+        for row in range(self.maze.grid_shape[0]):
+            for col in range(self.maze.grid_shape[1]):
+                # Draw node
+                img[
+                    row * unit_length + 1 : (row + 1) * unit_length,
+                    col * unit_length + 1 : (col + 1) * unit_length,
+                ] = node_color
+
+                # Down connection
+                if self.maze.connection_list[0, row, col]:
+                    img[
+                        (row + 1) * unit_length,
+                        col * unit_length + 1 : (col + 1) * unit_length,
+                    ] = connection_color
+
+                # Right connection
+                if self.maze.connection_list[1, row, col]:
+                    img[
+                        row * unit_length + 1 : (row + 1) * unit_length,
+                        (col + 1) * unit_length,
+                    ] = connection_color
+
+        return img
+
+    def _rowcol_to_coord(self, points: CoordArray) -> NDArray:
+        """transform points to img coordinates"""
```

Comment:
nit: I think this could be more descriptive - e.g. "Transform points from MazeTransformer to matplotlib default coordinate system"

Reply:
true, now: Transform Points from MazeTransformer (row, column) notation to matplotlib default (x, y) notation where x is the horizontal axis.